rfcs/0004-page-transitions.md

```diff
@@ -0,0 +1,207 @@
+- Start Date: 2020-01-13
+- Project: [next-with-moxy](https://github.com/moxystudio/next-with-moxy)
+- Reference Issues: N/A
+- Implementation PR: N/A
+
+# Summary
+
+This proposal is to add full page transitions to the `next-with-moxy` boilerplate, either as a guide and easy to use feature (recipe) or an out-of-the-box feature (by including it in the source code).
+
+# Basic example
+
+We will use [Highway](https://highway.js.org/) by Dogstudio, a lightweight library to create the navigation animations.
+
+## Installation
+
+First, install the package:
+
+```sh
+$ npm install --save @dogstudio/highway
+```
+
+## Enabling Highway
+
+In our [`App.js`](https://github.com/moxystudio/next-with-moxy/blob/master/www/app/App.js) file, we'll have to enable Highway:
+
+```js
+import Highway from '@dogstudio/highway/src/highway';
+import { Slide, Fade } from '../shared/components/transitions';
+
+class App extends NextApp {
+    Highway;
+
+    componentDidMount() {
+        this.unregisterGoogleTracking = registerGoogleTracking(this.props.router);
+
+        this.Highway = new Highway.Core({
+            transitions: {
+                '/contact': Fade,
```

Comment:
Is it possible to decide which animation to use depending on the current page? e.g.: When I'm on the homepage and click "Contact", I want the fade in animation to play; however, if I'm on the about page and click "Contact", I want the slide animation to play instead.

Reply:
I haven't tested it myself but I think so, with [contextual transitions](https://highway.js.org/get-started.html#contextual-transitions):

```js
const H = new Highway.Core({
	renderers: {
		name: CustomRenderer
	},
	transitions: {
		name: CustomTransition,
		contextual: {
			foo: ContextualTransition
		}
	}
});
```